api/cl_khr_kernel_clock.asciidoc

```diff
@@ -0,0 +1,64 @@
+// Copyright 2024 The Khronos Group Inc.
+// SPDX-License-Identifier: CC-BY-4.0
+
+include::{generated}/meta/{refprefix}cl_khr_kernel_clock.txt[]
+
+=== Other Extension Metadata
+
+*Last Modified Date*::
+    2024-03-25
+*IP Status*::
+    No known IP claims.
+*Contributors*::
+  - Kevin Petit, Arm Ltd. +
+  - Paul Fradgley, Imagination Technologies +
+  - Jeremy Kemp, Imagination Technologies +
+  - Ben Ashbaugh, Intel +
+  - Balaji Calidas, Qualcomm Technologies, Inc. +
+  - Ruihao Zhang, Qualcomm Technologies, Inc.
+
+=== Description
+
+`cl_khr_kernel_clock` adds the ability for a kernel to sample the value from one
+of three clocks provided by compute units.
+
+OpenCL C compilers supporting this extension will define the extension macro
+`cl_khr_kernel_clock`, and may define corresponding feature macros
+{opencl_c_kernel_clock_scope_device},
+{opencl_c_kernel_clock_scope_work_group}, and
+{opencl_c_kernel_clock_scope_sub_group} depending on the reported
+capabilities.
+
+See the link:{OpenCLCSpecURL}#cl_khr_kernel_clock[Kernel Clock] section of the
+OpenCL C specification for more information.
+
+=== Interactions With Other Extensions
+
+On devices that implement the `EMBEDDED` profile, the `cles_khr_int64` extension
+is required for the `clock_read_device`, `clock_read_work_group` and
+`clock_read_sub_group` functions to be present.
+
+Support for sub-groups is required for the `clock_read_sub_group` and
+`clock_read_hilo_sub_group` functions to be present.
+
+// The 'New ...' section can be auto-generated
+
+=== New Types
+
+  * {cl_device_kernel_clock_capabilities_khr_TYPE}
+
+=== New Enums
+
+  * {cl_device_info_TYPE}
+  ** {CL_DEVICE_KERNEL_CLOCK_CAPABILITIES_KHR}
+  * {cl_device_kernel_clock_capabilities_khr_TYPE}
+  ** {CL_DEVICE_KERNEL_CLOCK_SCOPE_DEVICE_KHR}
+  ** {CL_DEVICE_KERNEL_CLOCK_SCOPE_WORK_GROUP_KHR}
+  ** {CL_DEVICE_KERNEL_CLOCK_SCOPE_SUB_GROUP_KHR}
+
+include::provisional_notice.asciidoc[]
+
+=== Version History
+
+  * Revision 1.0.0, 2024-03-25
+  ** Initial version
```

Comment:
Should say something about being "(provisional)"? Other extension version histories do.

Reply:
Fixed.